src/psyclone/psyir/transformations/arrayrange2loop_trans.py

```diff
@@ -35,56 +35,62 @@
 # Modified by J. Henrichs, Bureau of Meteorology
 # Modified by A. B. G. Chalk, STFC Daresbury Lab
 
-'''Module providing a transformation from a PSyIR Array Range to a
-PSyIR Loop. This could be useful for e.g. performance reasons, to
-allow further transformations e.g. loop fusion or if the back-end does
+'''Module providing a transformation from a PSyIR Array Assignment to explicit
+PSyIR Loops. This could be useful for e.g. performance reasons, to enable
+further transformations e.g. loop fusion or if the back-end does
 not support array ranges.
 
-By default the transformation will reject character arrays,
-though this can be overriden by setting the
-allow_string option to True. Note that PSyclone expresses syntax such
-as `character(LEN=100)` as UnsupportedFortranType, and this
-transformation will convert unknown or unsupported types to loops.
-
 '''
 
+from psyclone.errors import LazyString
 from psyclone.psyGen import Transformation
-from psyclone.psyir.nodes import ArrayReference, Assignment, Call, \
-    IntrinsicCall, Loop, Literal, Range, Reference
-from psyclone.psyir.symbols import DataSymbol, INTEGER_TYPE, ScalarType, \
-        UnresolvedType, UnsupportedType, ArrayType, NoType
+from psyclone.psyir.nodes import (
+    Assignment, Call, IntrinsicCall, Loop, Literal, Range, Reference,
+    CodeBlock, Routine, BinaryOperation)
+from psyclone.psyir.nodes.array_mixin import ArrayMixin
+from psyclone.psyir.symbols import (
+    DataSymbol, INTEGER_TYPE, ScalarType, UnresolvedType, SymbolError,
+    ArrayType)
 from psyclone.psyir.transformations.transformation_error \
     import TransformationError
+from psyclone.psyir.transformations.reference2arrayrange_trans import (
+    Reference2ArrayRangeTrans)
 
 
 class ArrayRange2LoopTrans(Transformation):
     '''Provides a transformation from a PSyIR Array Range to a PSyIR
```

Comment:
Yes, this is always the always the use case we have.

I was wondering if I should change the name to ArrayRanges2Loops, or ArrayAssingment2Loops (since really it applies to the assignment)?

Reply:
Yes, let's go with `ArrayAssignment2Loops`.